packages/autocomplete-core/src/utils.ts

```diff
@@ -44,14 +45,25 @@ function normalizeSource<TItem>(
   };
 }
 
+const runConcurrentSafePromiseForGetSources = createConcurrentSafePromise<
+  Array<AutocompleteSource<any>>
+>();
+const runConcurrentSafePromiseForSource = createConcurrentSafePromise<
+  InternalAutocompleteSource<any>
+>();
+
 export function getNormalizedSources<TItem>(
   getSources: AutocompleteOptions<TItem>['getSources'],
   options
 ): Promise<Array<InternalAutocompleteSource<TItem>>> {
-  return Promise.resolve(getSources(options)).then((sources) =>
+  return runConcurrentSafePromiseForGetSources(
+    Promise.resolve(getSources(options))
```

Comment:
in both cases we are running it with Promise.resolve for the argument. I wonder if it makes sense to move that to createConcurrentSafePromise (before calling `.then`) so that it accepts sync & async functions natively. Wdyt?

Reply:
Done in 9fc796bce65fdb894f1770239254c5ab3d5f3942.